src/rules.ts

```diff
@@ -435,6 +438,18 @@ export class RuleChain extends LogicRule {
   }
 }
 
+export class RuleRaceChain extends RuleChain {
+  constructor(rule: RuleChain) {
+    if (0) {
+      super([])
+    }
+    rule.isBreak = RuleRaceChain.isBreak
+    return rule
+  }
+
+  static isBreak = (res: any) => res === true
```

Comment:
I think we should add a `resolve` method here. Ideally, `race` rule would run all rules, but resolve as soon as one of them returns a truthy result. Perhaps you could solve that using a parent callback and run the `Promises`?